src/CodeModel/Extensions/WebApi/UrlExtensions.cs

```diff
@@ -143,22 +150,36 @@ private static string ConvertRouteParameters(string route)
 
         private static string AppendQueryString(Method method, string route)
         {
-            // Todo: Add support for FromUri attribute
-
-            //var httpMethod = method.HttpMethod();
-            //if (httpMethod == "get" || httpMethod == "head")
-            //{
+            var httpMethod = method.HttpMethod();
             var prefix = route.Contains("?") ? "&" : "?";
 
-            foreach (var parameter in method.Parameters.Where(p => p.Type.IsPrimitive && p.Attributes.Any(a => a.Name == "FromBody") == false))
+            // The parameter list. This way we can filter out later:
+            IQueryable<Parameter> mParams = method.Parameters.AsQueryable();
+
+            if (httpMethod == "get" || httpMethod == "head")
+            {
+                // Filter the params list for the parameters that don't have "FromBody" attribute
+                mParams = from p in mParams
```

Comment:
This causes an incorrect URI generation in case if attribute parameter is not set. 
`mParams = mParams.Where(p => p.Type.IsPrimitive && !p.Attributes.Any(a => a.Name == "FromBody"));`
Works for me